src/ruby/puppetserver-lib/puppet/server/http_client.rb

```diff
@@ -169,37 +175,44 @@ def self.configure_metrics(client_options)
     end
   end
 
-  def self.create_client_options
+  def self.create_client_options(include_system_store = false)
     client_options = ClientOptions.new
     self.configure_timeouts(client_options)
-    self.configure_ssl(client_options)
+    self.configure_ssl(client_options, include_system_store)
     self.configure_metrics(client_options)
     client_options.set_enable_url_metrics(false)
     client_options
   end
 
-  def self.create_client
-    client_options = create_client_options
+  def self.create_client(include_system_store = false)
+    client_options = create_client_options(include_system_store)
     SyncHttpClient.createClient(client_options)
   end
 
   def self.client
     @client ||= create_client
   end
 
-  def self.client_post(request_options)
-    self.client.post(request_options)
+  def self.client_with_system_certs
+    @client_with_system_certs ||= create_client(:include_system_store)
```

Comment:
I'm confused about why `:include_system_store` is a keyword here / where its value is coming from..

Reply:
Fair, this is probably an antiquated Rubyism that predates keyword args or great tooling. It uses the fact that a symbol is a truthy value to give some semantic value to a `true` parameter. eg, instead of just writing `create_client(true)` it names what the true is conveying. I'll just convert to a kwarg.

Reply:
ahhh, gotcha, yeah I think a kwarg would be clearer.